File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;

        unordered_map<int, int> two_sum;

        for (int num : nums) {
            if (two_sum.contains(target - num)) {
                return {two_sum[target - num], i};
            }
            two_sum[num] = i;
            i += 1;
        }
    }
};

File: Data Structures & Algorithms/minimum-number-of-operations-to-make-array-empty/submission-3.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:

        num_count = Counter(nums)
        res = 0
        for num, count in num_count.items():
            if count == 1:
                return -1
            res+=math.ceil(count/3)
        return res
        